packages/core/src/Utils.ts

```diff
@@ -169,102 +189,199 @@ export function endsWith(input: string, suffix: string): boolean {
   return input.indexOf(suffix, input.length - suffix.length) !== -1;
 }
 
-// @ts-expect-error TS6133
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export function stringify(data: unknown, exclusions?: string[], maxDepth?: number): string {
-  function stringifyImpl(obj: unknown, excludedKeys: string[]): string {
-    const cache: unknown[] = [];
-    return JSON.stringify(obj, (key: string, value: unknown) => {
-      if (isMatch(key, excludedKeys)) {
-        return;
+/**
+ * This function will prune an object to a certain depth and return a new object.
+ * The following rules will be applied:
+ * 1. If the value is null or undefined, it will be returned as is.
+ * 2. If the value is a function or unsupported type it will be return undefined.
+ * 3. If the value is an array, it will be pruned to the specified depth.
+ * 4. If the value is an object, it will be pruned to the specified depth and
+ *    a. If the object is a Circular Reference it will return undefined.
+ *    b. If the object is a Map, it will be converted to an object. Some data loss might occur if map keys are object types as last in wins.
+ *    c. If the object is a Set, it will be converted to an array.
+ *    d. If the object contains prototype properties, they will be picked up.
+ *    e. If the object contains a toJSON function, it will be called and it's value will be normalized.
+ *    f. If the object is is not iterable or cloneable (e.g., WeakMap, WeakSet, etc.), it will return undefined.
+ *    g. If a symbol property is encountered, it will be converted to a string representation and could overwrite existing object keys.
+ * 5. If the value is an Error, we will treat it as an object.
+ * 6. If the value is a primitive, it will be returned as is.
+ * 7. If the value is a Regexp, Symbol we will convert it to the string representation.
+ * 8. BigInt and other typed arrays will be converted to a string unless number type works.
+ * 9. All other values will be returned as undefined (E.g., Buffer, DataView, Promises, Generators etc..)
+ */
+export function prune(value: unknown, depth: number = 10): unknown {
+  function isUnsupportedType(value: unknown): boolean {
+    if (value === null || value === undefined) {
+      return false;
+    }
+
+    switch (typeof value) {
+      case "function":
+        return true;
+      case "object":
+        switch (Object.prototype.toString.call(value)) {
+          case "[object AsyncGenerator]":
+          case "[object Generator]":
+          case "[object ArrayBuffer]":
+          case "[object Buffer]":
+          case "[object DataView]":
+          case "[object Promise]":
+          case "[object WeakMap]":
+          case "[object WeakSet]":
+            return true;
+        }
+
+        // Check for buffer;
+        if ("writeBigInt64LE" in value) {
+          return true;
+        }
+
+        break;
+    }
```

Comment:
Sure seems like we'd be better off having a list of things that we allow vs trying to keep an exhaustive list of unsupported types when new types can be added and probably different runtimes have different built in types as well. There are only so many intrinsic types that we should support.

Reply:
Yeah, we can work towards improving this. I'm assuming there will be more types over time, and we will need to optimize this some.